XCUITests/AuthenticationTest.swift

```diff
@@ -228,104 +187,108 @@ class AuthenticationTest: BaseTestCase {
-        navigator.goto(PasscodeSettings)
-        app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"].tap()
-        waitforExistence(app.staticTexts["Enter a passcode"])
-        typePasscode("133777")
+        navigator.performAction(Action.SetPasscodeTypeOnce)
         waitforExistence(app.staticTexts["Re-enter passcode"])
-        typePasscode("233777")
+
+        // Enter a passcode that does not match
+        userState.newPasscode = "333333"
+        navigator.performAction(Action.SetPasscodeTypeOnce)
         waitforExistence(app.staticTexts["Passcodes didn’t match. Try again."])
         waitforExistence(app.staticTexts["Enter a passcode"])
         app.buttons["Cancel"].tap()
     }
 
     func testPasscodeMustBeCorrectWhenRemoving() {
-        enablePasscode("133777")
+        navigator.performAction(Action.SetPasscode)
         XCTAssertTrue(app.staticTexts["Immediately"].exists)
-        app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode Off"].tap()
-        waitforExistence(app.staticTexts["Enter passcode"])
-        typePasscode("233777")
+
+        userState.passcode = "222222"
+        navigator.performAction(Action.DisablePasscode)
 
         waitforExistence(app.staticTexts["Incorrect passcode. Try again (Attempts remaining: 2)."])
-        typePasscode("133777")
+        userState.passcode = "111111"
+        navigator.nowAt(DisablePasscodeSettings)
+        navigator.performAction(Action.DisablePasscode)
         waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"])
     }
 
     func testChangingIntervalResetsValidationTimer() {
-        enablePasscode("133777")
+        navigator.performAction(Action.SetPasscode)
 
         // Enter login, since the default is 'set immediately,' it will ask for passcode
         navigator.goto(LoginsSettings)
         waitforExistence(app.navigationBars["Enter Passcode"])
-        typePasscode("133777")
+        navigator.performAction(Action.LoginPasscodeTypeOnce)
         waitforExistence(app.tables["Login List"])
 
-        navigator.goto(PasscodeIntervalSettings)
-        waitforExistence(app.navigationBars["Enter Passcode"])
-        typePasscode("133777")
-        waitforExistence(app.staticTexts["Immediately"])
-        app.staticTexts["After 15 minutes"].tap()
-
+        // Change it to 15 minutes
         navigator.goto(PasscodeSettings)
-        waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["After 15 minutes"])
+        setInterval(interval: "After 15 minutes")
 
         // Enter login, since the interval is reset, it will ask for password again
         navigator.goto(LoginsSettings)
         waitforExistence(app.navigationBars["Enter Passcode"])
-        disablePasscode("133777")
+        if iPad() {
+            app.buttons["Cancel"].tap()
+            navigator.nowAt(SettingsScreen)
+        }
```

Comment:
Should definitely go in to a back action in `LoginsSettings`. `isTablet` is available for you there.

Reply:
👍 I have added it there